test/test_lib/test_in_out/test_saver.py

```diff
@@ -0,0 +1,91 @@
+import unittest
+import lib.in_out.saver as saver
+import lib.util as util
+import lib.config as config
+import os
+import csv
+import json
+import networkx as nx
+
+current_directory = os.path.dirname(os.path.realpath(__file__))
+
+class SaverTest(unittest.TestCase):
+    
+    def setUp(self):
+        pass
+        
+    def tearDown(self):
+        pass
+
+    def test_check_if_dir_exists(self):
+        saver.check_if_dir_exists(current_directory + '/test/')
```

Comment:
@kaushik-rohit 
This directory was not setup in this test, but is deleted on line-23. That would not be an ideal unit test where there is a clean setup and teardown. Am I missing something here?

Reply:
check_if_dir_exists creates the directory if it doesn't exist.

Reply:
@kaushik-rohit 
In that case, wouldn't we want to have a test for the case of an already existing directory?

Reply:
Incase of already existing directory the method would do nothing. What would be a unit-test in such a case? perhaps checking if it wrongly created a directory?